app/lib/request.rb

```diff
@@ -94,9 +94,16 @@ def http_client
   class Socket < TCPSocket
     class << self
       def open(host, *args)
-        address = IPSocket.getaddress(host)
-        raise Mastodon::HostValidationError if PrivateAddressCheck.private_address? IPAddr.new(address)
-        super address, *args
+        outer_e = nil
+        Addrinfo.foreach(host, nil, nil, :SOCK_STREAM) do |address|
+          raise Mastodon::HostValidationError if PrivateAddressCheck.private_address? IPAddr.new(address.ip_address)
```

Comment:
Please move this line in the `begin...rescue` section. Such a case that one of addresses are invalid but the others are valid should have been tolerated in releases prior to v2.3.0.